Dynavity/Dynavity/persistence/data-mapping/CanvasWithAnnotation.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+/**
+ Wrapper for Canvas and AnnotationCanvas to be stored together in the persistence layer.
+ */
+class CanvasWithAnnotation {
+    let canvas: Canvas
+    // Remove dependency of storage with PencilKit
+    let annotationCanvas: Data
```

Comment:
Should this store the AnnotationCanvas directly instead, then `CanvasWithAnnotation` can be in the model? Otherwise the View-Model would be interacting with this directly, which is in the `Data Mapper` section

Reply:
Yep okay I think we can store `AnnotationCanvas` directly. Thanks!

>Also, should CanvasWithAnnotation be in the model? Otherwise the View-Model would be interacting with this directly, which is in the Data Mapper section

Yeap that makes sense too, but then it would mean that the `Data Mapper` section is interacting with the model?

Reply:
Nope I mean if we define the `CanvasWithAnnotation` class in the model, and we use this class in the `Data Mapper`, then that would mean that the `Data Mapper` is interacting with the model?

Perhaps I misunderstood your question on "Also, should CanvasWithAnnotation be in the model? Otherwise the View-Model would be interacting with this directly, which is in the Data Mapper section". My interpretation is that you're saying that the `CanvasWithAnnotation` class definition should be in the `model` directory instead of the `persistence` directory?

Reply:
>  My interpretation is that you're saying that the CanvasWithAnnotation class definition should be in the model directory instead of the persistence directory?

Yep, this is what I meant!

> Nope I mean if we define the CanvasWithAnnotation class in the model, and we use this class in the Data Mapper, then that would mean that the Data Mapper is interacting with the model?

I think this is fine? All the other DTOs are sort of interacting with their corresponding model. So in this case, the mapping is between `CanvasDTO` <-> `CanvasWithAnnotation`

Reply:
Also, now that I think about it, I'm not sure if we should store `Data` in the DTOs since that's quite `Codable`-specific. I think the DTOs should be independent of how we actually store the DTOs. But I think this is fine if there are no other alternatives

Reply:
Right makes sense, okay thanks!